src/TreeHouse/BehatCommon/PDOContext.php

```diff
@@ -0,0 +1,248 @@
+use PHPUnit_Framework_Assert as Assert;
+
+class PDOContext extends AbstractPersistenceContext
+{
+    /**
+     * @var string
+     */
+    private $dsn;
+
+    /**
+     * @var string
+     */
+    private $username;
+
+    /**
+     * @var string
+     */
+    private $password;
+
+    /**
+     * @var array
+     */
+    private $options;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @param string      $dsn
+     * @param string      $username
+     * @param string      $password
+     * @param array       $options
+     * @param string|null $purge_database_tag
+     */
+    public function __construct($dsn, $username, $password, array $options = [], $purge_database_tag = null)
+    {
+        $this->dsn = $dsn;
+        $this->username = $username;
+        $this->password = $password;
+        $this->options = $options;
+
+        parent::__construct($purge_database_tag);
+    }
+
+    /**
+     * @param string $name
+     * @param array  $data
+     */
+    protected function persistData($name, array $data)
+    {
+        $table = $this->convertNameToTable($this->singularify($name));
+
+        $this->persistRows($table, $data);
+    }
+
+    /**
+     * @param string $name
+     * @param array  $data
+     */
+    protected function assertDataPersisted($name, array $data)
+    {
+        $table = $this->convertNameToTable($this->singularify($name));
+
+        $this->assertRowsHaveBeenPersisted($table, $data);
+    }
+
+    /**
+     * @Then /^the following (.*?) should NOT exist:$/
+     *
+     * {@inheritdoc}
+     */
+    protected function assertDataNotPersisted($name, array $data)
+    {
+        $alias = $this->convertNameToTable($this->singularify($name));
+
+        $this->assertRowsHaveNotBeenPersisted($alias, $data);
+    }
+
+    /**
+     * @param string $table
+     * @param array  $data
+     *
+     * @throws DBALException
+     */
+    protected function insert($table, array $data)
+    {
+        $conn = $this->getConnection();
+
+        try {
+            $parameters = [];
+
+            foreach ($data as $key => $value) {
+                $parameters[$key] = ':' . $key;
+            }
+            $conn->beginTransaction();
+            $sql = sprintf(
+                'INSERT INTO `%s` (`%s`) VALUES (%s)',
+                $table,
+                implode('`,`', array_keys($parameters)),
+                implode(',', $parameters)
+            );
+
+            $stmt = $conn->prepare($sql);
+            $stmt->execute($data);
+
+            $conn->commit();
+        } catch (DBALException $e) {
+            $conn->rollBack();
+
+            throw $e;
+        }
+    }
+
+    /**
+     * @param string $table
+     * @param array  $criteria
+     *
+     * @return array|null
+     */
+    protected function find($table, array $criteria)
+    {
+        $parameters = [];
+        $conn = $this->getConnection();
+        $sql = sprintf('SELECT * FROM `%s` WHERE 1', $table);
```

Comment:
Mja die faalt dan weer als er geen criteria is (`WHERE ?`), maar met wat checks kan ik dat wel voorkomen, ga ik fixen